src/deploy.py

```diff
@@ -71,7 +71,9 @@ def _deploy():
     # BB8 restore
     data_exists = (not is_first_time) and service.settings["persist_data"]
     if settings["initial_data_source"] == "bb8_restore":
-        if data_exists:
+        data_update = service.settings["update_on_deploy"] and \
+                      not service.settings["bb8_backup"]
+        if data_exists and not data_update:
             print("Skipping bb8 restore: 'persist_data' is set, "
                   "and this is not a first-time deployment")
```

Comment:
This message should be updated. Ideally it should tell which half of the `and` predicate failed, so the user has a clear idea of what they need to change if this behaviour was not expected.

e.g. "Skipping bb8 restore: bb8_backup is set to True"

Reply:
Actually, this logic is quite complicated. Maybe we are conflating two things here. Maybe we should remove bb8 from the `initial_data_source` options, as I think it's not needed if we have this `update_on_deploy` mechanism. bb8 restore already happens at a different point in the process.

Reply:
Agree the message should be updated, but the logic seems simple enough and correct to me - if it's a first time deploy, or if `update_on_deploy` is true (and this isn't the source of truth for bb8, i.e. production), then run `bb8 restore`.

Reply:
But is there any need for "initial_data_source" to be bb8? Doesn't the `update_on_deploy` mechanism supersede that?

Reply:
What about for a first time deployment?  This should differ for a new support and production instance

Reply:
Okay fair enough. So if we're doing a clean install of production then we need initial_data_source to be bb8, but we don't want then to update every deploy. Whereas on support/science we want update_on_deploy.